actions/inventory.go

```diff
@@ -119,12 +119,39 @@ func Collect(ctx context.Context, device *common.Device, collectors *Collectors,
 		return errors.Wrap(err, "error retrieving TPM inventory")
 	}
 
+	if dynamic {
+		// Update StorageControllerCollectors
+		for _, sc := range device.StorageControllers {
+			switch common.FormatVendorName(sc.Vendor) {
```

Comment:
I suggest moving this utility detection into `utils/dynamic.go` as a method,

```go
CollectorBySlugVendor(componentSlug, vendor string) collectors.StorageControllers {
  switch componentSlug {
     case common.SlugStorageController:
          switch common.FormatVendorName(vendor) {
             case common.VendorMarvell:
          }
  }
}
```

Reply:
that function, as you defined it returns a var that implements collectors.StorageControllers -- should that be something that references collectors generically, since we take a componentSlug as the first param and this function could then return a collector for something other than StorageController?

Reply:
Thats a good point, although since currently theres no generic collector type, it is simpler to have methods returning the specific type instead. I should have named that method `StorageControllerCollector instead`.